templates/full/deviceFuelDrop.vm

```diff
@@ -4,6 +4,8 @@
 <body>
 Device: $device.name<br>
 Time: $dateTool.format("YYYY-MM-dd HH:mm:ss", $event.eventTime, $locale, $timezone)<br>
+Quantity Before: $!event.attributes.before<br>
```

Comment:
What does the exclamation point do here?

Reply:
It was supposed to prevent error if null was returned, zero is used instead